### DISCUSSION TITLE
Graphics capabilities

### DISCUSSION THREAD

**Discussion by brentward** [picocomputer/community#17] [Q&A]
Great job on this! I built a Ben Eater breadboard 6502 a while back and I added a bunch of stuff to it and got Ehbasic going through serial on a 65c51. I had the idea to use a Pico as a VGA card for it but I moved on when I didn't think it had enough IO lines to pull off well.

I'm curious how the graphics capabilities of this compare to the Vera module in the commander x 16. I'm not sure if there is anyone who actually has experience with both but the guys working on the commander x 16 say the Vera can come close to the graphics capabilities of the SNES but not quite match it. Does anyone have a ballpark of what this can pull off in comparison any particular retro systems?

**Comment by rumbledethumps** [comment]
The VERA does everything it needs to for 8-bit-style graphics but it doesn't do affine transforms. The SNES is famous for its mode 7 (affine transform). The VERA has 4096 colors. The SNES has 32768 colors. If the VERA is close to an SNES then the Picocomputer is close to being a time-travelling assassin robot.

Don't misunderstand. The VERA is good for its intended purpose. It's just not close to an SNES in my opinion.

**Reply by brentward** [reply]
I'm trying to figure out the what the capabilities of the VGA system will be in the new design in the VGA branch. It looks like the highest color mode is 256 colors, but the DAC seems like it is 5 bits per channel so will the overall colorspace be 32768 colors? It also will have affine transforms so based on my understanding it seems like it will  be more capable than the NES for example but it won't be able to match what the SNES could do (but maybe come closer in at least some respects than the VERA). Is that right?

How close are you to merging the vga branch into main?

**Comment by rumbledethumps** [comment]
https://www.youtube.com/watch?v=1n_4kE2SPt0
Looks like they came through with at least a partial screen of 3D. It's not clear how they did this, but the 6502 @ 8MHz is likely running the show here. There's some amazing programmers doing incredible things out there. Also, you can pre-order a bare-bones version for $350.

The SNES still beats both systems. A 16-bit ALU is more than twice as fast as 8-bit ALUs.

Has anyone seen parallax scrolling demos of the X16? I'm sure it's possible with sprites and palette rotations. What I'm most interested in is how approachable programming these systems is. Also, how will the skills you learn and refine transfer to modern programming.
